src/homework4_.py

```diff
@@ -0,0 +1,64 @@
+# 1 Создайте декоратор, который хранит в себе результаты вызовов функции
+# и время получения результата за время запуска программы.
+import time
+
+global reclist
+reclist = []
+global call_number
+call_number = 0
+
+
+def eto_moy_decorator(func):
+    """Декорирование функций с разным набором входных параметров"""
+    def obertochka(*args, **kwargs):
+        global call_number
+        call_number += 1
+        print('Вызов', call_number)
+        print('Входные параметры', args, kwargs)
+        start_time = time.time()
+        end_time = time.time()
+        result_of_call = ('Итог = ' + str(func(*args, **kwargs)) + '  ' +
+                          'Время вызова = ' + str(end_time - start_time))
+        reclist.append(result_of_call)
+
+    return obertochka
+
+
+@eto_moy_decorator
+def square(number=3):
+    """Возведение в квадрат с задержкой"""
+    result = pow(number, 2)
+    return result
+
+
+for input_number in range(6):
+    square(input_number)
+    print(reclist[input_number])
+
+
+# 2 Реализовать функцию get_ranges которая получает на вход непустой список
+# неповторяющихся целых чисел, отсортированных по возрастанию,
+# которая этот список “сворачивает”
+def get_ranges(input_list):
```

Comment:
['2-3', '6', '8-11', '13', '17-18']
['0-4', '7-8', '10']
['0-4', '7-8', '10']
['-4', '7', '10']
['2-3', '8-9']

is output of this function but should be single string